src/Json.php

```diff
@@ -103,6 +103,8 @@ public static function encode($valueToEncode, $cycleCheck = false, $options = []
                 return $valueToEncode->toJson();
             } elseif (method_exists($valueToEncode, 'toArray')) {
                 return static::encode($valueToEncode->toArray(), $cycleCheck, $options);
+            } elseif ($valueToEncode instanceof ArrayIterator) {
```

Comment:
ArrayInput have a transitive implementation of Iterator interface which implements Traversable. So this could be more generic and use iterator_to_array for transform any Traversable object